### PR TITLE
Feature/36387 check validator

### PR DESCRIPTION
**Pull request by GuyHarwood**
- [x] implement ts-jest with test change reload ability
- [x] rework async storage helper in Typescript
- [x] create mockable implementation to support TDD
- [x] implement Check Validator

### CODE REVIEW

File: func-complete-check/package.json [hunk 1]
```diff
@@ -4,31 +4,33 @@
...
+    "build": "yarn clean && tsc",
     "build:production": "yarn prestart",
     "watch": "tsc --w",
     "prestart": "yarn build && func extensions install",
     "start:host": "func start",
     "start": "yarn start:host & yarn watch",
-    "test": "yarn build && ./node_modules/.bin/jasmine",
-    "lint": "standard && tslint --project ."
+    "lint": "standard && tslint --project .",
+    "test": "yarn jest",
```
Comment (jon-shipley) [inline func-complete-check/package.json]: Do we want to explicitly show how to get a code coverage report from jest?  E.g `yarn test --coverage`

Comment (GuyHarwood) [reply func-complete-check/package.json]: will add 🏆

Comment (GuyHarwood) [reply func-complete-check/package.json]: done

File: func-complete-check/check-validator/check-validator.v1.ts [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+  receivedCheckTable: Array<any>
+  checkMarkingQueue: Array<any>
+}
+
+export class CheckValidatorV1 implements ICheckValidator {
+  private _tableService: IAsyncTableService
+  private _compressionService: ICompressionService
+
+  constructor (tableService?: IAsyncTableService, compressionService?: ICompressionService) {
+    if (tableService !== undefined) {
```
[the remark this answers is not in the document]
Comment (GuyHarwood) [reply func-complete-check/check-validator/check-validator.v1.ts]: undefined is guaranteed when the parameter is not provided.
https://www.typescriptlang.org/docs/handbook/functions.html#optional-and-default-parameters

File: func-complete-check/check-validator/check-validator.v1.ts [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+      this._tableService = tableService
+    } else {
+      this._tableService = new AsyncTableService()
+    }
+
+    if (compressionService !== undefined) {
```
[the remark this answers is not in the document]
Comment (GuyHarwood) [reply func-complete-check/check-validator/check-validator.v1.ts]: as above

File: func-complete-check/check-validator/check-validator.v1.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (GeorgeChatzigiannis) [reply func-complete-check/check-validator/check-validator.v1.ts]: good one!

**Pull request merged**